### PR TITLE
Update Cairo, add TestCompilationConfig, disable deprecated cairo-doc impl

### PR DESCRIPTION
**Pull request by maciektr**
- **Update cairo**
- **Disable deprecated cairo-doc code**
- **Add TestCompilationConfig to test compiler**
- **Change version to 2.6.4**
- **Add new edition**

### CODE REVIEW

**Review by THenry14 — APPROVED** [review]

File: extensions/scarb-doc/src/types.rs [hunk 1]
```diff
@@ -147,14 +146,15 @@ pub struct ItemData {
...
-        }
+    pub fn new(_db: &dyn DefsGroup, _id: ModuleItemId, _node: &impl TypedSyntaxNode) -> Self {
+        // Self {
+        //     name: id.name(db).into(),
+        //     full_path: id.full_path(db),
+        //     doc: db.get_item_documentation(LookupItemId::ModuleItem(id)),
+        //     definition: db.get_item_signature(LookupItemId::ModuleItem(id)),
+        //     text: node.as_syntax_node().get_text_without_trivia(db.upcast()),
+        // }
+        todo!("TODO(piotmag769): fix")
```
Comment (piotmag769) [inline extensions/scarb-doc/src/types.rs:157]: 👍

File: scarb/src/compiler/compilers/test.rs [hunk 2]
```diff
@@ -40,7 +40,13 @@ impl Compiler for TestCompiler {
...
         let test_compilation = {
             let _ = trace_span!("compile_test").enter();
-            compile_test_prepared_db(db, starknet, main_crate_ids, test_crate_ids)?
+            let config = TestsCompilationConfig {
+                starknet,
+                add_statements_functions: unit
+                    .compiler_config
+                    .unstable_add_statements_functions_debug_info,
+            };
+            compile_test_prepared_db(db, config, main_crate_ids, test_crate_ids)?
```
Comment (piotmag769) [inline scarb/src/compiler/compilers/test.rs:49]: 👍

File: scarb/tests/build.rs [hunk 3]
```diff
@@ -1108,3 +1108,88 @@ fn add_statements_functions_debug_info() {
...
+            .into_v1()
+            .unwrap()
+            .debug_info
+            .expect("Expected debug info to exist")
+            .annotations
+            .get("github.com/software-mansion/cairo-profiler")
+            .expect("Expected cairo-profiler annotations to exist")
+            .get("statements_functions")
+            .expect("Expected statements_functions info to exist")
+            .is_object(),
```
Comment (piotmag769) [inline scarb/tests/build.rs:1192]: [nit] Better to try to deserialize
https://github.com/software-mansion/scarb/blob/d32b7638ed6fcbb707f29482930b59bcbe5464af/extensions/scarb-snforge-test-collector/tests/test.rs#L473

Comment (maciektr) [reply scarb/tests/build.rs:1192]: Done in: #1407

**Pull request merged**